util_codes/View.py

```diff
@@ -0,0 +1,1128 @@
+
+        elif normal =="Z" or normal =="z":
+            self.x, self.y, self.z = x, y, z
+            self.ncx, self.ncy, self.ncz = x.size, y.size, 1
+            self.X, self.Y = np.meshgrid(x, y)
+            self.xyz = np.c_[self.X.flatten(), self.Y.flatten(), z*np.ones(self.ncx*self.ncy)]
+
+    def eval_loc(self, srcLoc,obsLoc, sigvec, fvec, orientation, func):
+        self.srcLoc=srcLoc
+        self.obsLoc = obsLoc
+        self.sigvec = sigvec
+        self.fvec = fvec
+        self.orientation = orientation
+        self.func = func
+        self.val_xfs=np.zeros((len(sigvec),len(fvec)),dtype=complex)
+        self.val_yfs=np.zeros((len(sigvec),len(fvec)),dtype=complex)
+        self.val_zfs=np.zeros((len(sigvec),len(fvec)),dtype=complex)
+   
+        for n in range(len(sigvec)):
+            #for k in range(len(fvec)):
+                self.val_xfs[n],self.val_yfs[n],self.val_zfs[n]=func(self.obsLoc, srcLoc, sigvec[n], fvec, orientation=self.orientation)
+    
+
+    def eval_2D(self, srcLoc, sig, f, orientation, func):
+        self.val_x, self.val_y, self.val_z = func(self.xyz, srcLoc, sig, f, orientation=orientation)
+        if self.normal =="X" or self.normal=="x":
+            Freshape = lambda v: v.reshape(self.ncy, self.ncz)
+            self.VAL_X, self.VAL_Y, self.VAL_Z = Freshape(self.val_x), Freshape(self.val_y), Freshape(self.val_z)
+            self.VEC_R_amp = np.sqrt(self.VAL_Z.real**2+self.VAL_Y.real**2)
+            self.VEC_I_amp = np.sqrt(self.VAL_Z.imag**2+self.VAL_Y.imag**2)
+            self.VEC_A_amp = np.sqrt(np.abs(self.VAL_Z)**2+np.abs(self.VAL_Y)**2)
+            self.VEC_P_amp = np.sqrt(phase(self.VAL_Z)**2+phase(self.VAL_Y)**2)
+
+        elif self.normal =="Y" or self.normal =="y":
+            Freshape = lambda v: v.reshape(self.ncx, self.ncz)
+            self.VAL_X, self.VAL_Y, self.VAL_Z = Freshape(self.val_x), Freshape(self.val_y), Freshape(self.val_z)
+            self.VEC_R_amp = np.sqrt(self.VAL_X.real**2+self.VAL_Z.real**2)
+            self.VEC_I_amp = np.sqrt(self.VAL_X.imag**2+self.VAL_Z.imag**2)
+            self.VEC_A_amp = np.sqrt(np.abs(self.VAL_X)**2+np.abs(self.VAL_Z)**2)
+            self.VEC_P_amp = np.sqrt(phase(self.VAL_X)**2+phase(self.VAL_Z)**2)
+
+        elif self.normal =="Z" or self.normal =="z":
+            Freshape = lambda v: v.reshape(self.ncx, self.ncy)
+            self.VAL_X, self.VAL_Y, self.VAL_Z = Freshape(self.val_x), Freshape(self.val_y), Freshape(self.val_z)
+            self.VEC_R_amp = np.sqrt(self.VAL_X.real**2+self.VAL_Y.real**2)
+            self.VEC_I_amp = np.sqrt(self.VAL_X.imag**2+self.VAL_Y.imag**2)
+            self.VEC_A_amp = np.sqrt(np.abs(self.VAL_X)**2+np.abs(self.VAL_Y)**2)
+            self.VEC_P_amp = np.sqrt(phase(self.VAL_X)**2+phase(self.VAL_Y)**2)
+
+
+    def plot2D_FD_RI(self, view="vec", ncontour=20, logamp=True, clim=None,showcontour=True, cont_perc=0.75):
```

Comment:
in fact, I would suggest pulling out the elements necessary to plot a single panel, and then call it twice - once with real, once with imag